File: 0958_Check Completeness of a Binary Tree/CheckCompletenessofaBinaryTree.cs
public class Solution {
    public bool IsCompleteTree(TreeNode root) {
        if(root == null) return true;
        
        var queue = new Queue<TreeNode>();
        queue.Enqueue(root);
        var missing = false;
        
        while(queue.Count > 0)
        {
            var size = queue.Count;
            while(size-- > 0)
            {
                var node = queue.Dequeue();    
                if(node == null)
                {
                    missing = true;
                }
                else
                {
                    if(missing) return false;
                    queue.Enqueue(node.left);
                    queue.Enqueue(node.right);
                }
            }
        }
        
        return true;
    }
}

File: 0958_Check Completeness of a Binary Tree/CheckCompletenessofaBinaryTree.java
class Solution {
    public boolean isCompleteTree(TreeNode root) {
        if(root == null) return true;
        Queue<TreeNode> queue = new LinkedList();
        queue.add(root);
        boolean missing = false;
        while(queue.size() > 0) {
            int size = queue.size();
            while(size-- > 0) {
                TreeNode tmpNode = queue.poll();
                if(tmpNode == null) {
                    missing = true;
                }else {
                    if(missing) return false;
                    queue.add(tmpNode.left);
                    queue.add(tmpNode.right);
                }
            }
        }
        
        return true;
    }
}